lightning/src/ln/channelmanager.rs

```diff
@@ -1334,6 +1340,57 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref> ChannelMan
 		})
 	}
 
+	#[allow(dead_code)]
+	// Messages of up to 64KB should never end up more than half full with addresses, as that would
+	// be absurd. We ensure this by checking that at least 500 (our stated public contract on when
+	// broadcast_node_announcement panics) of the maximum-length addresses would fit in a 64KB
+	// message...
+	const HALF_MESSAGE_IS_ADDRS: u32 = ::std::u16::MAX as u32 / (msgs::NetAddress::MAX_LEN as u32 + 1) / 2;
+	#[deny(const_err)]
+	#[allow(dead_code)]
+	// ...by failing to compile if the number of addresses that would be half of a message is
+	// smaller than 500:
+	const STATIC_ASSERT: u32 = Self::HALF_MESSAGE_IS_ADDRS - 500;
+
+	/// Generates a signed node_announcement from the given arguments and creates a
+	/// BroadcastNodeAnnouncement event. Note that such messages will be ignored unless peers have
+	/// seen a channel_announcement from us (ie unless we have public channels open).
+	///
+	/// RGB is a node "color" and alias is a printable human-readable string to describe this node
+	/// to humans. They carry no in-protocol meaning.
+	///
+	/// addresses represent the set (possibly empty) of socket addresses on which this node accepts
+	/// incoming connections. These will be broadcast to the network, publicly tying these
+	/// addresses together. If you wish to preserve user privacy, addresses should likely contain
+	/// only Tor Onion addresses.
+	///
+	/// Panics if addresses is absurdly large (more than 500).
+	pub fn broadcast_node_announcement(&self, rgb: [u8; 3], alias: [u8; 32], addresses: Vec<msgs::NetAddress>) {
+		let _ = self.total_consistency_lock.read().unwrap();
+
+		if addresses.len() > 500 {
+			panic!("More than half the message size was taken up by public addresses!");
+		}
```

Comment:
Good target for whomever addresses this issue https://github.com/rust-bitcoin/rust-lightning/issues/529